src/FoKL/FoKLRoutines.py

```diff
@@ -1695,115 +1701,109 @@ def clear(self, keep=None, clear=None, all=False):
+        ni_ids = []  # orders of basis functions used (where 0 is B1), per term
+        basis_n = []  # for future use when indexing 'm.fokl_basis'
+        for j in range(lv):  # for input variable in input variables
+            ni_ids.append(np.sort(np.unique(t[:, j][t[:, j] != -1])).tolist())
+            basis_n += ni_ids[j]
+        n_ids = np.sort(np.unique(basis_n))
 
-        try:
-            b = copy.deepcopy(self.betas_avg)
-        except:
-            b = np.mean(self.betas, axis=0)
-            self.betas_avg = b
+        if self.kernel != self.kernels[1]:
+            raise ValueError("The method for the 'Cubic Splines' kernel has not yet been ported from development, nor "
+                             "is expected to be as the resulting symbolic expressions are too infeasible to solve. Use "
+                             "the 'to_pyomo' method with a FoKL model trained on the 'Bernoulli Polynomials' kernel.")
 
-        t = self.mtx - 1  # indices of splines (where -1 means none)
-        lt = t.shape[0]  # length of terms (not including beta0)
-        lv = t.shape[1]  # length of input variables
-        s_ids = np.sort(np.unique(t[t != -1]))  # basis functions used for FoKL model
-
-        if self.kernel == self.kernels[0]:
-            lp = len(self.phis[0][0])  # length of cubics per spline (499)
-            if TruncateBasis is not None:  # assume TruncateBasis < lp
-                phinds64 = np.round(np.linspace(0, TruncateBasis - 1, TruncateBasis) * (lp - 1) / (TruncateBasis - 1))
-                phinds = []  # list of Python integers indexing the spline coefficients
-                for phind64 in phinds64:
-                    phinds.append(int(phind64))
-                p = []
-                for s in range(int(s_ids[-1] + 1)):  # for spline id in length of maximum spline id
-                    if s in s_ids:
-                        p4 = []
-                        for c in range(4):  # for x^power coefficient of powers 0 to 3
-                            p4.append(list(f.phis[s][c][phind] for phind in phinds))  # update splines needed for model
-                        p.append(p4)
-                    else:
-                        p.append([])  # spline not used, but still need to maintain indexing
-                p = tuple(p)  # to align with 'self.phis' format
-                lp = TruncateBasis  # = len(p[0][0])
+        if m is None:
+            m = pyo.ConcreteModel()
 
-                raise ValueError("The method for the 'Cubic Splines' kernel has not yet been ported from development.")
+        m.fokl_scenarios = pyo.Set(initialize=range(draws))  # index for scenario (i.e., FoKL draw)
+        m.fokl_y = pyo.Var(m.fokl_scenarios, within=pyo.Reals)  # FoKL output
 
+        m.fokl_j = pyo.Set(initialize=range(lv))  # index for FoKL input variable
+        m.fokl_x = pyo.Var(m.fokl_j, within=pyo.Reals, bounds=[0, 1])  # FoKL input variables (per domain)

```

Comment:
You can also try changing the initialization strategy here

Reply:
Couldn't get the correct solution this way but maybe is the same issue as `m.t = pyo.Var(..., initialize=0.5)`  finding the local max instead of global. Seems it will be desired for the initialization to take place within FoKLRoutines though instead of the example. Maybe we initialize `m.fokl_x` then look for another setting to solve for the global max? Not sure if possible

Reply:
resolved because IPOPT is meant to look for *local*, so functioning as expected